### PR TITLE
Rybeks Ungodly Ass Jam Changes

### PR DESCRIPTION
**Pull request by ryebek**
[ASS-JAM] [FEAT]
## About the PR 
Ass jam is supposed to be a shift from chaos through no rules, to chaos through ingame mechanics. So I decided to do what I thought would create the most chaos. HoS has a 5% chance to spawn as a macho man, captain has a 5% chance to spawn as tommy wiseau, the ammotech is replaced with a "Kinetitech" full of things that security isn't allowed to use, so will probably end up being used more by traitors, theres 300 meatcube slots, no HoS whitelist and its priority, and podseekers gib again. I've also tweaked the ratio of nuke ops to crew to reflect the crews new power, so its 1 nuke op for every 2 crew, with a max of 30.

## Why's this needed? 
Well chaos is fun, and this gonna cause a lot of chaos. Also lets us see what happens if HoS whitelist were to be removed, so we get to see some interesting Heads of Security. Jeffery Jefferson I'm counting on you.

## Changelog

```
(u)TrustworthyFella:
(*)Gave the HoS a 5% chance to be macho man Randy Savage
(*)Gave the captain a 5% chance to be Tommy Wiseau
(*)Brought back podseeker gibbing
(*)Removed HoS whitelist
(*)Made HoS a priority job
(*)Replaced the Ammotech with a Kinetitech, which is filled with lethals and stuff security can't actually use.
(*)Made the ratio of nuke ops to crew 1:2 rather than 1:4, and raised the maximum number of nuke ops to 30.
(*)Adds 300 latejoin meatcube slots, which are able to be traitors. 
(+)Meatcubes are announced when they join, like heads of staff are.
(+)Lets Heads of Security change their name when they spawn
```

### CODE REVIEW

File: code/datums/crew_objective.dm [hunk 1]
```diff
@@ -118,11 +118,12 @@
 						if(owner.current && !isdead(owner.current) && istype(get_area(owner.current),map_settings.escape_centcom)) //split this up as it was long
 							return 1
 				return 0
+
 		brigstir
 			explanation_text = "Keep Monsieur Stirstir brigged but also make sure that he comes to absolutely no harm."
 			medal_name = "Monkey Duty"
 			check_completion()
-				for(var/mob/living/carbon/human/npc/monkey/stirstir/M in mobs)
+				for(var/mob/living/carbon/human/M in mobs)
```
Comment (Tarmunora) [inline code/datums/crew_objective.dm:126]: Is this test-code that you accidentally left in?

File: code/datums/jobs.dm [hunk 2]
```diff
@@ -144,6 +144,14 @@
 		..()
 		src.access = get_all_accesses()
 
+	special_setup(var/mob/living/carbon/human/M)
+		..()
+		if (!M)
+			return
+		if(prob(5*ASS_JAM))
```
Comment (Tarmunora) [inline code/datums/jobs.dm:152]: Better to wrap things in #if/#else/#endif than to use the define directly in the code, for ease of searching and clearly telling what things do

File: code/datums/jobs.dm [hunk 3]
```diff
@@ -247,6 +261,9 @@
 		M.traitHolder.addTrait("training_drinker")
 		M.traitHolder.addTrait("training_security")
 		JOB_XP(M, "Head of Security", 1)
+		if(prob(ASS_JAM * 5))
```
Comment (Tarmunora) [inline code/datums/jobs.dm:266]: Again, should be wrapped in #if/#endif block

File: code/datums/jobs.dm [hunk 4]
```diff
@@ -2189,8 +2208,12 @@
 /datum/job/special/meatcube
 	name = "Meatcube"
 	linkcolor = "#FF0000"
-	limit = 0
+	limit = 0 + (ASS_JAM * 300)
+#if ASS_JAM
```
Comment (Tarmunora) [inline code/datums/jobs.dm:2221]: Again, the change to limit could have easily been in the #if/#else/#endif. Additionally, does this make meatcube valid for traitor? This is perhaps not wise.

File: code/modules/projectiles/bullet.dm [hunk 5]
```diff
@@ -690,7 +690,7 @@ toxic - poisons
 		var/type_to_seek = /obj/critter/gunbot/drone //what are we going to seek
 		precalculated = 0
 		on_hit(atom/hit, angle, var/obj/projectile/P)
-			if (P.data || prob(10)) //maybe
+			if (P.data || prob(10 + (ASS_JAM * 90))) //maybe, still explodes on ass jam, I think? maybe?
```
Comment (Tarmunora) [inline code/modules/projectiles/bullet.dm:693]: #if/etc/etc

File: code/modules/vending/vending.dm [hunk 6]
```diff
@@ -408,32 +408,66 @@
...
 	create_products()
 		..()
-		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/abg, 6)
-		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/a38, 2)
-		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/a38/stun, 3)
-		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/flare, 3)
-		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/smoke, 3)
-		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/tranq_darts, 3)
-		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/tranq_darts/anti_mutant, 3)
+		product_list += new/datum/data/vending_product(/obj/item/ammo/bullets/abg, 6 * (ASS_JAM * 20))//on ass jam theres 20 times as much ammo in the ammotech
```
Comment (Tarmunora) [inline code/modules/vending/vending.dm:435]: Again, better to wrap in the #if/else/endif. Duplicate code for the sake of readability - making it explicitly clear what is and is not part of ass_jam content

File: code/datums/jobs.dm (hunk 4 above)
Comment (ryebek) [reply code/datums/jobs.dm:2221]: Meatcube is indeed valid for traitor, but its latejoin only, so you sort of know what you're getting into.

File: code/datums/crew_objective.dm (hunk 1 above)
Comment (ryebek) [reply code/datums/crew_objective.dm:126]: Its from a previous job objective I was persuaded to remove.

File: _stdlib/__build.dm [hunk 7]
```diff
@@ -112,7 +112,7 @@ var/global/ci_dm_version_minor = "100"
 #define BUILD_TIME_FULL "2009-02-13 18:31:30"
 #define BUILD_TIME_YEAR 2053
 #define BUILD_TIME_MONTH 01
-#define BUILD_TIME_DAY 01 //SET ME TO 13 TO TEST YOUR ASS_JAM CONTENT!!
+#define BUILD_TIME_DAY 13 //SET ME TO 13 TO TEST YOUR ASS_JAM CONTENT!!
```
Comment (Tarmunora) [inline _stdlib/__build.dm:115]: You don't need to include changes to this file

File: code/datums/crew_objective.dm (hunk 1 above)
Comment (Tarmunora) [reply code/datums/crew_objective.dm:126]: It appears to be changing an *existing* objective

File: code/datums/gamemodes/nuclear.dm [hunk 8]
```diff
@@ -34,7 +38,7 @@
 		if (player.client && player.ready)
 			num_players++
 
-	var/num_synds = max(1, min(round(num_players / 4), agents_possible))
+	var/num_synds = max(1, min(round(num_players / (4 / (ASS_JAM * 2)), agents_possible))
```
Comment (Tarmunora) [inline code/datums/gamemodes/nuclear.dm:41]: why. the #if/#else/#endif is recommended for a reason

File: code/datums/crew_objective.dm (hunk 1 above)
Comment (ryebek) [reply code/datums/crew_objective.dm]: Fixed. I think.

File: _stdlib/__build.dm (hunk 7 above)
Comment (ryebek) [reply _stdlib/__build.dm]: Fixed.

**Comment by jaffjafferson** [comment]
speedmerge this but remove all the ass jam checks

**Comment by pali6** [comment]
Please fix your changelog.

**Comment by ryebek** [comment]
Changes done, bugs fixed, I think thats everything.

File: code/modules/projectiles/bullet.dm [hunk 9]
```diff
@@ -693,7 +693,7 @@ toxic - poisons
 #if ASS_JAM
 			if (P.data || prob(100))) //Removing the data check would mean indenting is fucked, and im lazy
```
Comment (Tarmunora) [inline code/modules/projectiles/bullet.dm:694]: unbalanced parens. must fix

Comment (ryebek) [reply code/modules/projectiles/bullet.dm:694]: Yeah, thats fixed.

Comment (pali6) [reply code/modules/projectiles/bullet.dm:694]: it's not

**Pull request merged**